admin/atlas_client.go

```diff
@@ -58,24 +60,11 @@ func UseDigestAuth(apiKey, apiSecret string) ClientModifier {
 // credentials.LocalTokenCache can be supplied to reuse OAuth Token across application restarts.
 // Warning: for advanced use cases please use credentials.NewTokenSource directly in your code pass it to UseHTTPClient method.
 // Warning: any previously set httpClient will be overwritten. To fully customize HttpClient use UseHTTPClient method.
-func UseOAuthAuth(clientID, clientSecret string, tokenCache credentials.LocalTokenCache) ClientModifier {
+func UseOAuthAuth(clientID, clientSecret string, tokenCache *oauth2.Token) ClientModifier {
 	return func(c *Configuration) error {
-		var tokenSource credentials.RevocableTokenSource
-		if tokenCache != nil {
-			tokenSource = credentials.NewTokenSourceWithOptions(credentials.AtlasTokenSourceOptions{
-				ClientID:     clientID,
-				ClientSecret: clientSecret,
-				TokenCache:   tokenCache,
-				BaseURL:      &c.Servers[0].URL,
-			})
-		} else {
-			tokenSource = credentials.NewTokenSourceWithOptions(credentials.AtlasTokenSourceOptions{
-				ClientID:     clientID,
-				ClientSecret: clientSecret,
-				BaseURL:      &c.Servers[0].URL,
-			})
-		}
-		httpClient := credentials.NewHTTPClientWithOAuthToken(tokenSource)
+		oauth := credentials.NewConfig(clientID, clientSecret)
+		source := oauth2.ReuseTokenSource(tokenCache, oauth.TokenSource(context.Background()))
```

Comment:
[Q] Why using `ReuseTokenSource` from oauth2.

Reply:
because it has the logic to check if the token is valid and reuse it or request a new one and keep using it while valid, I don't think we need to implement that logic ourselves

Reply:
Prior to your commit, wrapper of ReuseToken source provided the ability to save OAuth token.

Reply:
yes for file systems, for in memory there's nothing to save, it just works

Reply:
Added method for saving token in filesystem